pandas/tests/generic/test_frame.py

```diff
@@ -280,3 +281,30 @@ def test_deepcopy_empty(self):
         empty_frame_copy = deepcopy(empty_frame)
 
         self._compare(empty_frame_copy, empty_frame)
+
+    def test_datetime_after_setitem_with_at(self):
+        # This test covers the unexpected behaviour of datetimeField when using
+        # setitem on another column as reported in issue #6942
+
+        start = pd.to_datetime("20140401")
+
+        result = pd.DataFrame(
+            index=pd.date_range(start, periods=1), columns=["timenow", "Live"]
+        )
+
+        result.at[start, "timenow"] = datetime.today()
+
+        new_datetime = datetime.today()
```

Comment:
does it matter that this datetime.today() is called separately from the one two lines up?  if not, pls use just one datetime object